src/hotspot/share/runtime/objectMonitor.cpp

```diff
@@ -595,6 +594,9 @@ bool ObjectMonitor::deflate_monitor() {
     install_displaced_markword_in_object(obj);
   }
 
+  // Release object's oop storage since the ObjectMonitor has been deflated:
+  release_object();
```

Comment:
Just to be clear, deflate_monitor() is called when a JavaThread is
in VM (or by the VM Thread) so it is safe to call release.

Reply:
I hope OopStorage.release() will assert if it's not safe.  I think it does.

Reply:
Yes it will assert() when it is called from a not safe context.